apps/server/src/modules/learnroom/controller/course.controller.ts

```diff
@@ -36,15 +36,23 @@ export class CourseController {
 	@Get(':courseId/export')
 	async exportCourse(
 		@CurrentUser() currentUser: ICurrentUser,
-		@Param('courseId') courseId: string,
+		@Param() urlParams: CourseUrlParams,
 		@Res({ passthrough: true }) response: Response
 	): Promise<StreamableFile> {
 		if (!this.configService.get<boolean>('FEATURE_IMSCC_COURSE_EXPORT_ENABLED')) throw new NotFoundException();
-		const result = await this.courseExportUc.exportCourse(courseId, currentUser.userId);
+		const result = await this.courseExportUc.exportCourse(urlParams.courseId, currentUser.userId);
 		response.set({
 			'Content-Type': 'application/zip',
 			'Content-Disposition': 'attachment;',
 		});
 		return new StreamableFile(result);
 	}
+
+	@Get(':courseId')
+	async getCourseForTeacher(@CurrentUser() currentUser: ICurrentUser, @Param() urlParams: CourseUrlParams) {
```

Comment:
@virgilchiriac I think it's a little bit confusing that this special function (..forTeacher) is executed for this usual endpoint "/{courseId)".

Reply:
could be that we rename it to getCourse, but then also we should have a way to handle if the user is student (non-teacher), e.g. throw an exception...

Reply:
it could only expose the attributes the user is allowed to see